applications/main/infrared/infrared_cli.c

```diff
@@ -333,57 +340,140 @@ static void infrared_cli_process_decode(Cli* cli, FuriString* args) {
-            infrared_brute_force_add_record(brute_force, i++, "MUTE");
-        } else if(strncmp(furi_string_get_cstr(command), "vup", 3) == 0) {
-            infrared_brute_force_add_record(brute_force, i++, "VOL+");
-        } else if(strncmp(furi_string_get_cstr(command), "vdown", 3) == 0) {
-            infrared_brute_force_add_record(brute_force, i++, "VOL-");
-        } else if(strncmp(furi_string_get_cstr(command), "cup", 3) == 0) {
-            infrared_brute_force_add_record(brute_force, i++, "CH+");
-        } else if(strncmp(furi_string_get_cstr(command), "cdown", 3) == 0) {
-            infrared_brute_force_add_record(brute_force, i++, "CH-");
-        } else {
+    do {
+        if(!args_read_string_and_trim(args, command)) {
+            printf("Invalid command please see help.\r\n");
             infrared_cli_print_usage();
+            break;
+        }
+
+        if(strncmp(furi_string_get_cstr(command), "list", 4) == 0) {
+            args_read_string_and_trim(args, remote);
+            if(strncmp(furi_string_get_cstr(remote), "tv", 2) == 0) {
+                Remote = TV;
+            } else if(strncmp(furi_string_get_cstr(remote), "ac", 2) == 0) {
+                Remote = AC;
+            } else {
+                printf("Invalid remote type.\r\n");
+                break;
+            }
+            infrared_cli_list_remote_signals(Remote);
+            break;
+        }
+
+        if(strncmp(furi_string_get_cstr(command), "tv", 2) == 0) {
+            Remote = TV;
+        } else if(strncmp(furi_string_get_cstr(command), "ac", 2) == 0) {
+            Remote = AC;
+        } else {
+            printf("Invalid remote type.\r\n");
+            break;
         }
 
-        bool success = infrared_brute_force_calculate_messages(brute_force);
-        if(success) {
-            printf("Sending codes to tv...\r\n");
+        args_read_string_and_trim(args, signal);
+        if(furi_string_empty(signal)) {
+            printf("Must supply a valid signal for type of remote selected.\r\n");
+            break;
+        }
 
-            uint32_t record_count;
-            uint32_t index = 0;
+        infrared_cli_brute_force_signals(cli, Remote, signal);
+        break;
 
-            bool running = infrared_brute_force_start(brute_force, index, &record_count);
+    } while(false);
 
-            while(running) {
-                running = infrared_brute_force_send_next(brute_force);
-            }
+    furi_string_free(command);
+    furi_string_free(remote);
+    furi_string_free(signal);
+}
+
+static void infrared_cli_list_remote_signals(enum RemoteTypes remote_type) {
+    Storage* storage = furi_record_open(RECORD_STORAGE);
+    FlipperFormat* ff = flipper_format_buffered_file_alloc(storage);
+    const char* remote_file = NULL;
+    bool success = false;
+
+    switch(remote_type) {
+    case TV:
+        remote_file = EXT_PATH("infrared/assets/tv.ir");
```

Comment:
Suggestion:
Instead of using an enum, build the file path with user-supplied string ("tv" or "ac"), with "list" being a reserved keyword for listing the available remotes (it's unlikely that we'll ever have a universal remote library called "list").
If there is no such file (the user failed to enter "ac" or "tv"), print an error message.
Current implementation is also okay.